pkg/controller/direct/directbase/directbase_controller.go

```diff
@@ -223,7 +247,6 @@ func (r *reconcileContext) doReconcile(ctx context.Context, u *unstructured.Unst
 		}
 		return false, r.handleUpdateFailed(ctx, u, err)
 	}
-	k8s.EnsureFinalizers(u, k8s.ControllerFinalizerName, k8s.DeletionDefenderFinalizerName)
```

Comment:
I'm now even more confused :-)  I agree this seems awfully late to be adding finalizers, but better late than never?

Reply:
I don't think we should be removing this line, but not a blocker as this controller is feature-flagged out